src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -76,37 +91,191 @@ static DamageTable D_809901F8 = {
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = EnTg_Idle;
+    this->actor.gravity = -4.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/func_8098FA70.s")
+void EnTg_Destroy(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/EnTg_Update.s")
+    Collider_DestroyCylinder(play, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/func_8098FBB4.s")
+/**
+ * The actor spins, and a heart is spawned above the actor every 12 frames.
+ */
+void EnTg_Idle(EnTg* this, PlayState* play) {
+    Vec3f heartStartPos;
+
+    this->actor.shape.rot.y += sREG(0) + 0x258;
+    this->actor.world.rot = this->actor.shape.rot;
+
+    if (DECR(this->spawnHeartTimer) == 0) {
+        this->spawnHeartTimer = 12;
+        heartStartPos = this->actor.world.pos;
+        heartStartPos.y += 62.0f;
+        EnTg_SpawnHeart(this, this->effects, &heartStartPos, ARRAY_COUNT(this->effects));
+    }
+}
+
+void EnTg_Update(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+
+    this->actionFunc(this, play);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnTg_UpdateSkelAnime(this, play);
+    EnTg_UpdateHearts(play, this->effects, ARRAY_COUNT(this->effects));
+    EnTg_UpdateCollider(this, play);
+}
+
+s32 EnTg_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    return false;
+}
+
+void EnTg_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnTg* this = THIS;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+
+    if (limbIndex == HONEY_AND_DARLING_LIMB_MAN_HEAD) {
+        Matrix_MultVec3f(&zeroVec, &this->actor.focus.pos);
+    }
+}
+
+void EnTg_Draw(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+    GraphicsContext* gfxCtx;
+
+    Matrix_Push();
+    EnTg_DrawHeart(play, this->effects, ARRAY_COUNT(this->effects));
+    Matrix_Pop();
+
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Gfx_EnvColor(play->state.gfxCtx, 0, 50, 160, 0));
+    gSPSegment(POLY_OPA_DISP++, 0x09, Gfx_EnvColor(play->state.gfxCtx, 255, 255, 255, 0));
+
+    SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                          EnTg_OverrideLimbDraw, EnTg_PostLimbDraw, &this->actor);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+/**
+ * Spawns a heart at the first effects array index that's not enabled.
+ * Because of the frame counts, only two hearts are ever spawned at a time.
+ */
+void EnTg_SpawnHeart(EnTg* this, EnTgHeartEffect* effects, Vec3f* heartStartPos, s32 len) {
+    Vec3f heartVelocityVec = { 0.0f, 1.5f, 0.0f };
```

Comment:
```suggestion
    Vec3f heartVelocity = { 0.0f, 1.5f, 0.0f };
```
Just velocity is fine, that implied it's a vector. You could use "speed" if it was a scalar